src/utils/isIosDevice.ts

```diff
@@ -0,0 +1,18 @@
+let isIos: boolean | undefined;
+function isIosDevice(): boolean {
+    if (isIos !== undefined) {
+        return isIos;
+    }
+
+    isIos = /iphone|ipad|ipod|macintosh/i.test(navigator.userAgent.toLowerCase()) && navigator.maxTouchPoints > 1;
+    return isIos;
+}
+
+function resetIosDeviceCachedValue() {
+    isIos = undefined;
+}
+
+export const iOs = {
+    isIosDevice,
+    resetIosDeviceCachedValue
```

Comment:
Why is this reset needed? 🤔 I would think once we know if the device is iOs or not, that won't change in the given session. So once this is set, why would we want to reset it's value? 🤔

Reply:
Agree with you, this is code smell pre-mature optimization. I'll delete it as it just complicates the code.